doc/specs/#147 - Support different release channels [released, beta, alpha].md

```diff
@@ -0,0 +1,259 @@
+```raw
+/manifests/7/7zip/7zip/1.0
+  | 7zip.7zip.installer.yaml
+  | 7zip.7zip.installer.Alpha.yaml
+  | 7zip.7zip.locale.en-US.yaml
+  | 7zip.7zip.yaml
+/manifests/7/7zip/7zip/2.0
+  | 7zip.7zip.installer.Alpha.yaml
+  | 7zip.7zip.locale.en-US.yaml
+  | 7zip.7zip.yaml
+```
+
+### CLI Behavior
+Where not specified by the user, the default release channel should be used/shown first or exclusively, then other channels in ASCII sort order. If the package manifest has a default channel specified and no versions of that package have an installer manifest matching the default channel, then the next available channel should be used/shown. Channels should be treated as *case-insensitive*.
+
+The channel `Default` shall be treated as a reserved token. When used in commands, default should refer to the default channel specified in the version manifest. In command output, when the default channel is named, the name should be shown. If the default channel is null, then `Default` should be shown. Manifest validation should throw an error when `DefaultChannel` is `Default`
+
+#### winget search
+In order to avoid package duplication in search results, only the default channel should be shown unless otherwise specified. Users should be able to search for packages by release channel in addition to all other search filters.
+
+Some valid searches:
+```powershell
+PS> winget search --channel <channel> # Return all packages which have a channel matching <channel>
+PS> winget search --channel "" # Return all packages which have a null channel
+PS> winget search <query> --channel <channel> # Return packages matching <query> which have a channel matching <channel>
+PS> winget search --tag <tag> --channel <channel> # Return packages with a tag matching <tag> which have a channel matching <channel>
+...
+```
+
+#### winget show
+When the user specifies `--versions`, the versions for all channels along with the channels should be shown. The versions should be grouped by channel, sorted by channel, then sorted by version descending
+When the user specifies `--channels`, the available channels for the package should be shown. If a package contains a named default channel and unnamed (null) channels, only the named channels should be shown
+
+#### winget install
+When the user does not specify `--channel`, the default channel should be used. If the user does specify `--channel`, the specified channel should be used. The latest version for the channel should be installed, unless a user specified the version using the `--version` parameter. Additionally, the Windows Package Manager should record the package channel in some manner to ensure that upgrades to the package are found using the correct channel.
+
+Optionally, the user should be able to specify the channel as part of the package identifier using the `@` symbol. For example, `winget install Google.Chrome@dev` would be equivalent to `winget install Google.Chrome --channel dev`
```

Comment:
Unfortunately we didn't reserve any characters for this kind of thing.  We could still support it, but it would probably need to check for packages named `Google.Chrome` and `Google.Chrome@dev` and then continue depending on what is found.

If we do decide to go down this route, we should also consider supporting other identifiers (well, mostly one, version).  Maybe `id@channel#version`?  Then we would probably want to reserve these on `winget-pkgs` even if the client needed to handle it all.